azurelinuxagent/ga/exthandlers.py

```diff
@@ -889,15 +889,15 @@ def __get_handlers_on_file_system(self, incarnation_changed):
 
         return handlers_to_report
 
-    def report_ext_handlers_status(self, incarnation_changed=False):
+    def report_ext_handlers_status(self, incarnation_changed=False, vm_agent_upgrade_status=None):
         """
         Go through handler_state dir, collect and report status.
         Returns the status it reported, or None if an error occurred.
         """
         try:
             vm_status = VMStatus(status="Ready", message="Guest Agent is running",
-                                 gs_aggregate_status=self.__gs_aggregate_status)
-
+                                 gs_aggregate_status=self.__gs_aggregate_status,
+                                 vm_agent_update_status=vm_agent_upgrade_status)
```

Comment:
NIT: these two have slightly different names (`...agent_update...` vs `...agent_upgrade...`), might consider picking just a single one.

Reply:
good catch! Let me fix it.